Yank/restraints.py

```diff
@@ -551,20 +698,295 @@ def _determineBondParameters(self):
             sigma = 1.4826 * median_absolute_distance
 
             # Calculate r0, which is a multiple of sigma plus 5 A.
-            r0 = 2*sigma + 5.0 * units.angstroms
+            r0 = 2*sigma + 5.0 * unit.angstroms
         else:
-            DEFAULT_DISTANCE = 15.0 * units.angstroms
+            DEFAULT_DISTANCE = 15.0 * unit.angstroms
             logger.warning("Receptor only contains %d atoms; using default distance of %s" % (natoms, str(DEFAULT_DISTANCE)))
             r0 = DEFAULT_DISTANCE
 
-        logger.debug("restraint distance r0 = %.1f A" % (r0 / units.angstroms))
+        logger.debug("restraint distance r0 = %.1f A" % (r0 / unit.angstroms))
 
         # Set spring constant/
-        #K = (2.0 * 0.0083144621 * 5.0 * 298.0 * 100) * units.kilojoules_per_mole/units.nanometers**2
-        K = 0.6 * units.kilocalories_per_mole / units.angstroms**2
-        logger.debug("K = %.1f kcal/mol/A^2" % (K / (units.kilocalories_per_mole / units.angstroms**2)))
+        #K = (2.0 * 0.0083144621 * 5.0 * 298.0 * 100) * unit.kilojoules_per_mole/unit.nanometers**2
+        K = 0.6 * unit.kilocalories_per_mole / unit.angstroms**2
+        logger.debug("K = %.1f kcal/mol/A^2" % (K / (unit.kilocalories_per_mole / unit.angstroms**2)))
 
         # Assemble parameter vector.
         bond_parameters = [K, r0]
 
         return bond_parameters
+
+
+#=============================================================================================
+# Base class for orientation-dependent receptor-ligand restraints.
+#=============================================================================================
+
+class OrientationDependentRestraint(ReceptorLigandRestraint):
+    """
+    Impose an orientation-dependent restraint between ligand and protein to prevent ligand from
+    changing to a different binding mode.
+
+    This restraint strength is controlled by a global context parameter called 'lambda_restraints'.
+
+    """
+    energy_function = ''  # energy function to use in computation of restraint
+    bond_parameter_names = []  # list of bond parameters that appear in energy function above
```

Comment:
No, I've removed this.

We *do* still need a way to write these parameters to the NetCDF file.